### PR TITLE
Deploy applications using server-side apply

### PR DESCRIPTION
**Pull request by VaishnaviHire**
## Description
This commit makes use of `ServerSideApply` option provided by
kubectl package to perform server-side apply while deploying manifests.
This will fix issues that result in failed reconciliation when oversized
CRDs are deployed.

## How Has This Been Tested?
1. Deploy custom operator image using instructions given here : https://github.com/opendatahub-io/opendatahub-operator/blob/master/operator.md#build-instructions (My custom image: quay.io/vhire/kfctl:v1.2)
2. Test if following KfDefs are applied successfully:
     - [ODH](https://github.com/opendatahub-io/odh-manifests/blob/master/kfdef/kfctl_openshift.yaml)
     - [Notebook Controller](https://github.com/opendatahub-io/odh-manifests/blob/master/kfdef/kfctl_odh_notebook_controller.yaml)
     - [Kubeflow Manifests](https://github.com/opendatahub-io/manifests/blob/v1.5-branch-openshift/openshift/kfdef/kfctl_openshift_v1.5.0_odh.yaml) 
3. Update Notebook Controller KfDef and confirm if the operator reconciles.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by VaishnaviHire** [comment]
Review Request: @samuelvl @anishasthana

**Review by anishasthana — COMMENTED** [review]
/lgtm

I was able to apply multiple kfdefs without issue.

**Comment by VaishnaviHire** [comment]
@LaVLaS Did you get a chance to review this ?

**Comment by LaVLaS** [comment]
I'm testing this against the current ODH components just find any odh-manifests bugs that will be triggered by this update. Once this is merged, this will force a lot of updates to odh-manifests to unblock any ODH deployments that may be deploying using this feature

File: pkg/utils/k8utils.go
Comment (LaVLaS) [inline pkg/utils/k8utils.go:317]: Does this need to be `application/apply-patch+yaml`

**Review by LaVLaS — CHANGES_REQUESTED** [review]
Just one commet about the patchType but 
/lgtm

We'll have to do a full regression test with all of the valid components in odh-manifests to see what manifests updates will be required once this is active.  Most of the issues I've found were actual odh-manifests bugs OR dead resource fields that could be safely ignored but fail schema validation

File: pkg/utils/k8utils.go
Comment (VaishnaviHire) [reply pkg/utils/k8utils.go:317]: Its the expected content type for Yaml/JSON apply https://kubernetes.io/docs/reference/using-api/server-side-apply/#apply-and-update

Comment (VaishnaviHire) [reply pkg/utils/k8utils.go]: @LaVLaS Fixed this. Also updated the test image with this change

**Review by LaVLaS — APPROVED** [review]

**Pull request merged**
